datahub/opal.py

```diff
@@ -61,9 +109,17 @@ def create_opal_frame() -> pd.DataFrame:
     return df
 
 
-def append_opal_frame(data: list[float]) -> pd.DataFrame:
+def append_opal_frame(data: dict[str, float]) -> pd.DataFrame:
```

Comment:
This function doesn't actually append anything, and I'm suggesting it return a from from a DataFrame (which is a `pd.Series`):

```suggestion
def get_opal_row(data: dict[str, float]) -> pd.Series:  # type: ignore[type-arg]
```

Note: mypy needs to be ignored here because our setting do not allow "Generic Types" and a pd.Series is a generic type - mypy wants us to define the internal data type (ie `pd.Series[float]`) - but the rows contain that one Timestamp variable, so we need to ignore that.